src/gpodder/extensions.py

```diff
@@ -378,18 +379,41 @@ def _find_extensions(self):
             builtins = os.path.join(gpodder.prefix, 'share', 'gpodder',
                 'extensions', '*.py')
             user_extensions = os.path.join(gpodder.home, 'Extensions', '*.py')
-            self.filenames = glob.glob(builtins) + glob.glob(user_extensions)
 
-        # Let user extensions override built-in extensions of the same name
+            # sort filenames so that if duplicates are found in the same folder,
+            # the highest priority (lowest number) will always be used.
+            self.filenames = sorted(glob.glob(builtins), reverse=True) \
+                + sorted(glob.glob(user_extensions), reverse=True)
+
+        # Let user extensions override built-in extensions of the same name.
+        # This inherently happens because we search the user extensions folder second,
+        # and the entries are put in the extensions dict by their name field.
         for filename in self.filenames:
             if not filename or not os.path.exists(filename):
                 logger.info('Skipping non-existing file: %s', filename)
                 continue
 
             name, _ = os.path.splitext(os.path.basename(filename))
-            extensions[name] = filename
 
-        return sorted(extensions.items())
+            # extensions with no priority get lowest priority, 99
+            priority = 99
+            m = re.fullmatch(r'^([0-9]*)_(.+)', name)
+            if m:
+                if m.group(1):
+                    # get ordering prefix
+                    priority = int(m.group(1))
+                # strip ordering prefix from name (or leading _)
+                name = m.group(2)
+            if name in extensions:
+                if filename.startswith(gpodder.home) == extensions[name][1].startswith(gpodder.home):
+                    logger.warning("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
+                else:
+                    logger.info("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
```

Comment:
I'd suggest:

- Do query if old extension exists and retrieval of the name in a single command (also avoids `extensions[name][1]`, which one needs to look for what `[1]` actually contains)
- The "warn if in the same directory, info if different directory" logic is better understood with `dirname(...) == dirname(...)` vs `.startswith(home) == .startswith(home)` (also it would cover more cases if we ever source from more than two directories)

```suggestion
            _, previous_filename = extensions.get(name, (None, None))
            if previous_filename is not None:
                if os.path.dirname(gpodder.home) == os.path.dirname(previous_filename):
                    logger.warning("extension at %s will be ignored in favor of %s in the same directory", previous_filename, filename)
                else:
                    logger.info("extension at %s will be ignored in favor of %s", previous_filename, filename)
```

Reply:
Took me a minute but I think I see what you're saying, and looks fine to me. Committed.